File: sem3/map/Lab/3/toylanguage/src/main/java/com/company/Models/Statements/CompoundStatement.java
package com.company.Models.Statements;

import com.company.Models.IStatement;
import com.company.Utils.MyIStack;
import com.company.Models.ProgramState.ProgramState;

public class CompoundStatement implements IStatement {
    private IStatement firstStatement;
    private IStatement secondStatement;

    public CompoundStatement(IStatement firstStatement, IStatement secondStatement) {
        this.firstStatement = firstStatement;
        this.secondStatement = secondStatement;
    }


    @Override
    public ProgramState execute(ProgramState state) {
        MyIStack<IStatement> stack = state.getExeStack();

        stack.push(secondStatement);
        stack.push(firstStatement);

        return state;
    }

    @Override
    public String toString() {
        return firstStatement.toString() + ";" + secondStatement.toString();
    }
}

File: sem3/map/Lab/toylanguage_C#/ToyLanguage/ToyLanguage/Models/Statements/CompoundStatement.cs
using ToyLanguage.Models.Interfaces;
using ToyLanguage.Models.States;

namespace ToyLanguage.Models.Statements
{
    public class CompoundStatement : IStatement
    {
        private readonly IStatement _firstStatement;
        private readonly IStatement _secondStatement;

        public CompoundStatement(IStatement firstStatement, IStatement secondStatement)
        {
            _firstStatement = firstStatement;
            _secondStatement = secondStatement;
        }
        
        public ProgramState Execute(ProgramState state)
        {
            state.ExecutionStack.Push(_secondStatement);
            state.ExecutionStack.Push(_firstStatement);

            return state;
        }

        public override string ToString()
        {
            return $"{_firstStatement}\n{_secondStatement}";
        }
    }
}